go/vt/sqlparser/ast.go

```diff
@@ -2851,10 +2851,14 @@ type (
 
 	AggrFunc interface {
 		Expr
-		AggrName() string
 		GetArg() Expr
-		IsDistinct() bool
 		GetArgs() Exprs
+		// AggrName returns the lower case string representing this aggregation function
+		AggrName() string
+	}
+
+	DistinctableAggr interface {
+		IsDistinct() bool
```

Comment:
any reason to make `DistinctableAggr` as interface?

Reply:
seemed like a better design than what we had. do you have any alternatives in mind?